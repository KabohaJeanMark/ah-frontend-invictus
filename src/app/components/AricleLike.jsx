// import React from 'react';
// import PropTypes from 'prop-types';

// const ArticleLike = ({ handleLike, handleDislike }) => {
//   return (

//         <>
//         <div className="like-article">
//           <div className="row">
//             <div className="col-md-2" />
//             <div className="col-md-8">
//               <div className="card">
//                 <div className="like-dislike">
//                   <i
//                     className="far fa-thumbs-up"
//                     onClick={handleLike}
//                     role="presentation"
//                     style={{ color: 'blue' }
//                     }
//                   />
//                   <i
//                     className="far fa-thumbs-down"
//                     onClick={handleDislike}
//                     role="presentation"
//                     style={{ color: 'red' }}
//                   />
//                 </div>
//               </div>

//             </div>
//             <div className="col-md-2" />
//           </div>
//         </div>

//     </>
//   );
// };

// ArticleLike.propTypes = {
//   handleLike: PropTypes.func,
//   handleDislike: PropTypes.func
// };

// export default ArticleLike;
