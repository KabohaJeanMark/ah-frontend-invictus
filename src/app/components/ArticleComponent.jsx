import React from 'react';
import { WithContext as ReactTags } from 'react-tag-input';
import '../../assets/styles/article.scss';
import Navbar from '../components/Navbar';

const KeyCodes = {
  comma: 188,
  enter: 13
};

const delimiters = [KeyCodes.comma, KeyCodes.enter];

const ArticlePresentationComponent = ({ handleSubmit, handleChange, handleAddTag, handleDeleteTag, tags }) => {
  return (
    <>
    <Navbar />
      <div className="create-article">
        <div className="row">
          <div className="col-md-2" />
          <div className="col-md-8">
            <div className="card">
              <form onSubmit={handleSubmit}>
                <div className="form-group">
                  <label className="sr-only">Title</label>
                  <textarea
                    className="title-input"
                    type="text"
                    name="title"
                    id="title-input"
                    placeholder="Title"
                    required
                    onChange={(e) => handleChange(e)}/>
                </div>
                <div className="form-group">
                  <textarea
                    className="description-input"
                    type="text"
                    name="description"
                    id="description-input"
                    placeholder="Description"
                    required
                    onChange={(e) => handleChange(e)}
                  />
                </div>
                <div className="form-group">
                  <textarea
                    className="body-input"
                    type="text"
                    name="body"
                    id="body-input"
                    placeholder="Body"
                    required
                    onChange={e => handleChange(e)}
                  />
                </div>
                <div className="form-group">
                  <label htmlFor="tags">Tags</label>
                  <ReactTags
                    classNames="tags"
                    maxLength = "42"
                    inline
                    tags={tags}
                    handleAddition={handleAddTag}
                    handleDelete={handleDeleteTag}
                    delimiters={delimiters}
                  />
                </div>
                <button type="submit" className="btn postarticle-btn btn-outline-primary">Post Article</button>
              </form>
            </div>
          </div>
          <div className="col-md-2" />
        </div>
      </div>
    </>
  );
};

export default ArticlePresentationComponent;
