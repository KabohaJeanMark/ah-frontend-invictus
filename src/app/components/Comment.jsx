import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/login.scss';


const Comment = ({ handleSubmit, handleChange }) => {
  return (
    <div>
      <div className="row">
        <div className="col-md-5" />
        <form onSubmit={handleSubmit}>
          <label>Comment</label>
          <textarea id="CommentBody" onChange={handleChange}
            className="form-control input-field" name="commentBody"
            placeholder="Enter your Comments" />
          <input type="submit" className="btn btn-primary login-btn" />
        </form>
      </div>
    </div>
  );
};

Comment.propTypes = {
  handleSubmit: PropTypes.func,
  handleChange: PropTypes.func
};

export default Comment;
