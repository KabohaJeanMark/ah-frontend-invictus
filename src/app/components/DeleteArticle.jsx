import React from 'react';

export default function DeleteArticleComponent (props) {
  return (
    <div >
      <button className="btn btn-outline-danger"
        onClick={props.handleDelete}>Delete</button>
    </div>
  );
}
