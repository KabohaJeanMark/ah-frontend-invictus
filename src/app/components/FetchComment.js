import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/login.scss';
import '../../assets/styles/fetchcomment.scss';

const FetchComment = ({ comments }) => {
  return (
    <>
      <div className="row content">
        <div className="col-md-12">
          {
            comments.map(comment => {
              return (
                <div className="input-field" >
                  {comment.body}
                  <br />
                  {comment.created_at}
                  <br />
                  {comment.author}
                </div>
              );
            })
          }

        </div>
      </div>
    </>
  );
};

FetchComment.propTypes = {
  comment: PropTypes.string,
  comments: PropTypes.array
};

FetchComment.defaultProps = {
  comments: []
};

export default FetchComment;
