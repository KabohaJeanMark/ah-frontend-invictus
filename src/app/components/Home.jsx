import React, { Component } from 'react';
import NavbarComponent from '../components/Navbar';
import Articles from '../containers/articles/ArticlesContainer';

class Home extends Component {
  render () {
    return (
      <React.Fragment>
        <NavbarComponent />
        <Articles />
      </React.Fragment>
    );
  }
}

export default Home;
