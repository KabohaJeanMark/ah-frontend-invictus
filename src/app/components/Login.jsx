import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/styles/login.scss';

const Login = (
  {
    handleSubmit,
    onChange,
    email,
    password,
    errors
  }
) => {
  return (
    <div className="row">
      <div className="col-sm-4" />
      <div className="col-sm-4">
        <div className="banner">
          <img className="banner-img"
            src={require("../../assets/images/ah_banner.png")}
            alt={"Banner"} />
        </div>
        <div className="card">
          <form onSubmit={handleSubmit}>
            <div className="form-group">
              <label>Login Form</label>
              <input type="text" id="UserEmail" onChange={onChange} value={email || ''} name="email"
                className="form-control input-field" placeholder="Email" required />
              {
                errors.userEmail && (<div style={{ color: 'red' }}>{errors.userEmail}</div>)
              }
              <input type="password" onChange={onChange} value={password} name="password"
                className="form-control input-field" placeholder="Password" required />
              {
                errors.userPassword && (<div style={{ color: 'red' }}>{errors.userPassword}</div>)
              }
              <input type="submit" className="btn btn-primary login-btn" />
              <p className="have-account">Forgot password? <a href="/password/reset">Reset</a></p>
            </div>
            <div className="loginwith">
              <img className="loginwith-img" alt="login-with"
                src={require("../../assets/images/loginwith.png")} />
              <div className="row social-media-row">
                <div className="col-md-2" />
                <div className="col-md-8">
                  <img className="socialmedia-img facebook-img" alt="facebook-logo"
                    src={require("../../assets/images/facebook.png")} />
                  <img className="socialmedia-img google-img" alt="google-logo"
                    src={require("../../assets/images/google.png")} />
                  <img className="socialmedia-img twitter-img" alt="twitter-logo"
                    src={require("../../assets/images/twitter.png")} />
                </div>
                <div className="col-md-2" />
              </div>
            </div>
            <p className="have-account">Not yet a member? <a href="/signup">Sign up</a></p>
          </form>
        </div>
      </div>
      <div className="col-sm-3" />
    </div>
  );
};
Login.propTypes = {
  email: PropTypes.string,
  password: PropTypes.string,
  errors: PropTypes.object,
  handleSubmit: PropTypes.func,
  onChange: PropTypes.func
};

export default Login;
