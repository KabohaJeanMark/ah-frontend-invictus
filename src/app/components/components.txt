All app components will go into this folder.
Delete this after adding atleast one component.
