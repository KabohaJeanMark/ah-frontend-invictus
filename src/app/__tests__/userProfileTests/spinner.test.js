import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { SpinnerComponent } from '../../components/SpinnerComponent';

describe('<SpinnerComponent />', () => {
  let wrapper;
  it('it renders spinner without crashing', () => {
    wrapper = shallow(<SpinnerComponent />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
