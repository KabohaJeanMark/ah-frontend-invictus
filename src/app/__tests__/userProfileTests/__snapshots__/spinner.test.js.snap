// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SpinnerComponent /> matches snapshot 1`] = `
<div
  className="spinner-div"
>
  <div
    className="spinner-grow text-primary"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-grow text-secondary"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-grow text-success"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-grow text-danger"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-grow text-warning"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
  <div
    className="spinner-grow text-info"
    role="status"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </div>
</div>
`;
